packages/lit-html/src/lit-html.ts

```diff
@@ -706,6 +711,8 @@ class Template {
   }
 }
 
+type HasParent = NodePart | AttributePart | TemplateInstance;
```

Comment:
Add comment. Because it's not obvious, even with TypeScript's help, I'd highlight the common interface that's used.